### PR TITLE
Refactoring near azure blob storage

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

<details>
    <summary>Modify your CI run</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [x]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [x]  All with Azure
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

<details>

### CODE REVIEW

**Comment by CurtizJ** [comment]
The idea of refactoring is to unify creation of client in azure disk and in azure storage.

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit fa8b47d8dc6c0289fa486aeb83bf67ee37aa63fb with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63636/fa8b47d8dc6c0289fa486aeb83bf67ee37aa63fb/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/integration/test_backup_restore_azure_blob_storage/test.py [hunk 1]
```diff
@@ -30,12 +30,12 @@ def generate_cluster_def(port):
...
         <azure_conf1>
             <connection_string>DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://azurite1:{port}/devstoreaccount1;</connection_string>
-            <container>cont</container>
+            <container_name>cont</container_name>
             <format>CSV</format>
         </azure_conf1>
         <azure_conf2>
             <storage_account_url>http://azurite1:{port}/devstoreaccount1</storage_account_url>
-            <container>cont</container>
+            <container_name>cont</container_name>
```
Comment (kssenii) [inline tests/integration/test_backup_restore_azure_blob_storage/test.py:38]: May be support "container" as well? Should be a 1 line change, but will allow to not make backward incompatible changes

**Review by kssenii — APPROVED** [review]

File: tests/integration/test_backup_restore_azure_blob_storage/test.py (hunk 1 above)
Comment (CurtizJ) [reply tests/integration/test_backup_restore_azure_blob_storage/test.py]: Fixed.

**Comment by CurtizJ** [comment]
The PR failed: too many conflicts. Basically it requires to reimplement it from scratch.

File: src/Storages/ObjectStorage/Azure/Configuration.h [hunk 2]
```diff
@@ -54,22 +54,12 @@ class StorageAzureConfiguration : public StorageObjectStorage::Configuration
...
-    std::string container;
     std::string blob_path;
     std::vector<String> blobs_paths;
+    AzureBlobStorage::ConnectionParams connection_params;
 
-    AzureClientPtr createClient(bool is_read_only, bool attempt_to_create_container);
-    AzureObjectStorage::SettingsPtr createSettings(ContextPtr local_context);
-    Poco::URI getConnectionURL() const;
+    // AzureClientPtr createClient(bool is_read_only, bool attempt_to_create_container);
+    // AzureObjectStorage::SettingsPtr createSettings(ContextPtr local_context);
```
Comment (kssenii) [inline src/Storages/ObjectStorage/Azure/Configuration.h:62]: remove?

**Review by kssenii — APPROVED** [review]

**Comment by CurtizJ** [comment]
Test `00002_log_and_exception_messages_formatting` is failed in master with azure.

**Comment by clickgapai** [comment]
Hi @CurtizJ @kssenii — while reviewing this PR I found the following:

- **Bug (P1):** `BackupWriterAzureBlobStorage::copyFile` destination path missing `blob_path` prefix — https://github.com/ClickHouse/ClickHouse/pull/104445

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**